### PR TITLE
[WIP] Move to Java 11 as the minimum version to anticipate Lucene

### PR DESCRIPTION
**Pull request by idodeclare**
Upcoming Lucene 9.0.0 has updated its System Requirements with LUCENE-8738: Move to Java 11 as minimum Java version.

This is a patch to try a revised configuration to move OpenGrok to 11 too.

### CODE REVIEW

**Comment by tarzanek** [comment]
lgtm,
once this is in ... the next release should be blogged about that OpenGrok now requires jdk > 8

fwiw - I personally still use tomcat + jdk8
and opengrok + jdk8 
since ... it's stable

I am not sure how much testing went into jdk11 and tomcat
+ same on our side, is someone using jdk > 11 + opengrok? @ahornace @tulinkry ?

**Comment by tulinkry** [comment]
I'm using the docker image which runs on jdk11 in production and haven't seen any problems. On my computer I use jdk12 to run opengrok, but I haven't been very active recently to test it properly.

**Comment by vladak** [comment]
The instance I maintain internally is still on JDK 8 + Tomcat 8 and it might be a bit of a hurdle to jump to JDK 11 (due to Solaris x86 Java support - https://www.oracle.com/java/technologies/javase/products-doc-jdk11certconfig.html) however I feel going to JDK 11 as it is the next LSU is the way to go (JDK 14 deprecates even SPARC anyway - https://bugs.openjdk.java.net/browse/JDK-8231554).

**Comment by vladak** [comment]
>     * same on our side, is someone using jdk > 11 + opengrok? @ahornace @tulinkry ?

I have been using JDK 11 on Ubuntu for all my local OpenGrok work for quite a while, no problems there.

File: pom.xml [hunk 1]
```diff
@@ -326,16 +331,16 @@ Portions Copyright (c) 2018, 2020, Chris Fraire <cfraire@me.com>.
 
     <profiles>
         <profile>
-            <id>java9plus-8-release</id>
+            <id>java12plus-11-release</id>
```
Comment (vladak) [inline pom.xml]: should we just get rid of these Java 9+ profiles ? there are similar entries in the other modules, e.g. indexer passes extra args when running tests, or suggester adds extra modules.

Comment (idodeclare) [reply pom.xml]: I merged the `java9plus` profiles into the default, but this one here I believe is meant to set the `<release>` element to 11.

Comment (vladak) [reply pom.xml]: Hmm, okay.

**Comment by vladak** [comment]
needs rebase otherwise this is good to go from my point of view.

**Comment by idodeclare** [comment]
> needs rebase otherwise this is good to go from my point of view.

Maddeningly after much trial and error I never got the windows build to work on this branch.

I checked now on Windows 10 20xx and a different set of errors occurs. So 🤮

**Comment by vladak** [comment]
I see that the Suggester tests are failing currently. Perhaps this needs the changes in #3207 ?

**Comment by idodeclare** [comment]
Rebased and also included a commit for Java 14 Javadoc fixes when I was trying to resolve the build on Windows (with no success)

**Comment by ahornace** [comment]
> I checked now on Windows 10 20xx and a different set of errors occurs. So 🤮

I see that Windows build passed in Travis. Is OpenGrok + JDK8 working on Windows 10 20xx properly? Travis does not seem to support this version so basically any change can cause the build failure on this OS and we would be blind to it.

@vladak Cannot we wave it off? I think this is definitely a needed step.

**Comment by vladak** [comment]
Looks like the Windows version used for Travis build is 'windows server 2019'. Personally I am fine with whatever Windows version Travis runs on. I see the Linux/macOS checks failed because of failed download.

**Comment by idodeclare** [comment]
Unexpected coverage reduction of -11.4%. Not sure if due to intermittent build failure. Rebased to try again.

**Comment by vladak** [comment]
I restarted the builds in my branch and this time it is macOS build that is failing with:
```
Copying opengrok_tools-1.4.8-py3.7.egg to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/lib/python3.7/site-packages
Adding opengrok-tools 1.4.8 to easy-install.pth file
Installing opengrok script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-config-merge script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-deploy script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-groups script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-indexer script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-java script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-mirror script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-projadm script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-reindex-project script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installing opengrok-sync script to /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/bin
Installed /Users/travis/build/vladak/OpenGrok/opengrok-tools/target/env/lib/python3.7/site-packages/opengrok_tools-1.4.8-py3.7.egg

Processing dependencies for opengrok-tools==1.4.8
Searching for filelock
Reading https://pypi.org/simple/filelock/
Download error on https://pypi.org/simple/filelock/: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1076) -- Some packages may not be found!

Couldn't find index page for 'filelock' (maybe misspelled?)
```

I think this is hitting #2528 due to Xcode version bump.

**Comment by vladak** [comment]
The last macOS build was successful. The Linux failed because of a network problem. So, there is nothing that would be blocking this.

**Comment by vladak** [comment]
> The last macOS build was successful. The Linux failed because of a network problem. So, there is nothing that would be blocking this.

Pulled the latest branch and macOS build still fails due to setuptools vs. certificates.

**Comment by vladak** [comment]
Let's see if this holds up.

**Comment by vladak** [comment]
will update the wikis with the new Java version.

**Pull request merged**